# BRIEF PREDICTION DE LA GRAVITE DES ACCIDENTS DE LA ROUTE

## Import des librairies

In [4]:
import numpy as np
import pandas as pd

import plotly.express as px
import matplotlib.pyplot as plt
import seaborn as sns

## Lieux

### Import du jeu de données 

In [7]:
df_lieux = pd.read_csv("lieux-2023.csv", delimiter=';',encoding='utf-8') 
df_lieux

C:\Users\peggy\AppData\Local\Temp\ipykernel_25552\618727500.py:1: DtypeWarning: Columns (12) have mixed types. Specify dtype option on import or set low_memory=False.
  df_lieux = pd.read_csv("lieux-2023.csv", delimiter=';',encoding='utf-8')


Num_Acc  catr                                      voie  v1   v2  \
0      202300000001     4                             RUE DE RIVOLI   0  NaN   
1      202300000001     4                       RUE SAINT FLORENTIN   0  NaN   
2      202300000002     3                                       120   0  NaN   
3      202300000003     3                                         5   0  NaN   
4      202300000003     3                                        87   0  NaN   
...             ...   ...                                       ...  ..  ...   
70855  202300054818     4                 LA FONTAINE (RUE JEAN DE)   0  NaN   
70856  202300054819     4                      RN3 (ANCIENNE ROUTE)   0  NaN   
70857  202300054820     4              BAMBOU (CHEMIN DE LA RUELLE)   0  NaN   
70858  202300054821     2                         ROUTE NATIONALE 1   0  NaN   
70859  202300054822     4  YVES FARGES                         (BD)   0  NaN   

       circ nbv  vosp  prof   pr  pr1  plan lartpc larrout  surf  infra  situ  \
0         1   2     0     1   -1   -1     1    NaN      -1     2      0     1   
1         1   1     0     1   -1   -1     1    NaN      -1     2      0     1   
2         2   3     2     1   -1   -1     1    NaN      -1     2      0     1   
3         2   4     0     1    1    0     1    NaN      -1     2      5     1   
4         2   4     0     1    1    0     1    NaN      -1     2      5     1   
...     ...  ..   ...   ...  ...  ...   ...    ...     ...   ...    ...   ...   
70855     2   2     0     1   -1    0     1    NaN      -1     1      0     1   
70856     2   2     0     2    0    0     1    NaN      -1     1      0     1   
70857     2   2     0     2    0    0     2    NaN      -1     1      0     8   
70858    -1   2     1     1    4   50     1    NaN      -1     1      0     1   
70859     2   2     2     1   -1   -1     1    NaN      -1     1      0     5   

       vma  
0       30  
1       30  
2       50  
3       50  
4       50  
...    ...  
70855   50  
70856   50  
70857   30  
70858   90  
70859   30  

[70860 rows x 18 columns]

In [8]:
df_lieux.head(20)

Num_Acc  catr                       voie  v1   v2  circ nbv  vosp  \
0   202300000001     4              RUE DE RIVOLI   0  NaN     1   2     0   
1   202300000001     4        RUE SAINT FLORENTIN   0  NaN     1   1     0   
2   202300000002     3                        120   0  NaN     2   3     2   
3   202300000003     3                          5   0  NaN     2   4     0   
4   202300000003     3                         87   0  NaN     2   4     0   
5   202300000004     2                          6   0  NaN     2   4     0   
6   202300000005     4                        NaN   0  NaN     1   1     0   
7   202300000005     4                        NaN   0  NaN     2   2     0   
8   202300000006     4                        NaN   0  NaN     1   1     0   
9   202300000007     3                          7   0  NaN     2   4     3   
10  202300000008     3                        130   0  NaN     3   4     0   
11  202300000008     3                        NaN   0  NaN     2   2     0   
12  202300000009     3                          5   0  NaN     3   4     0   
13  202300000009     3                         87   0  NaN     1   2     0   
14  202300000010     3                        231   0  NaN     2   2     0   
15  202300000011     1        CHAPIT RE ( RUE DU)   0  NaN     1   5     0   
16  202300000012     1           DOCTEUR VAILLANT   0  NaN     1   1     0   
17  202300000012     1           DOCTEUR VAILLANT   0  NaN     4   2     0   
18  202300000013     3  CROIZAT (AVENUE AMBROISE)   0  NaN     2   2     0   
19  202300000014     3      LINDBERGH (CARREFOUR)   0  NaN     1   2     0   

    prof   pr    pr1  plan lartpc larrout  surf  infra  situ  vma  
0      1   -1     -1     1    NaN      -1     2      0     1   30  
1      1   -1     -1     1    NaN      -1     2      0     1   30  
2      1   -1     -1     1    NaN      -1     2      0     1   50  
3      1    1      0     1    NaN      -1     2      5     1   50  
4      1    1      0     1    NaN      -1     2      5     1   50  
5      1   18      1     1    NaN      12     2      0     1   50  
6      1   -1     -1     1    NaN      -1     2      0     1   30  
7      1   -1     -1     1    NaN      -1     2      0     1   30  
8      1   -1     -1     1    NaN      -1     2      0     1   50  
9      1   -1     -1     1    NaN      -1     2      0     1   50  
10     1   -1     -1     1    NaN      -1     2      0     1   50  
11     1   -1     -1     1    NaN      -1     2      0     1   50  
12     1   -1     -1     1    NaN      -1     2      5     1   50  
13     1   -1     -1     1    NaN      -1     1      5     1   50  
14     1   -1     -1     1    NaN      -1     2      0     1   50  
15     1   18  1 630     1    NaN      -1     2      3     1   70  
16     2   14      0     1    NaN      -1     1      3     1   50  
17     2    0      0     1    NaN      -1     1      0     1   50  
18     2    0      1     1    NaN       3     2      0     1   50  
19     1    0      1     1    NaN      -1     2      0     1   50

In [9]:
df_lieux.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 70860 entries, 0 to 70859
Data columns (total 18 columns):
 #   Column   Non-Null Count  Dtype 
---  ------   --------------  ----- 
 0   Num_Acc  70860 non-null  int64 
 1   catr     70860 non-null  int64 
 2   voie     58113 non-null  object
 3   v1       70860 non-null  int64 
 4   v2       5884 non-null   object
 5   circ     70860 non-null  int64 
 6   nbv      70860 non-null  object
 7   vosp     70860 non-null  int64 
 8   prof     70860 non-null  int64 
 9   pr       70860 non-null  object
 10  pr1      70860 non-null  object
 11  plan     70860 non-null  int64 
 12  lartpc   31 non-null     object
 13  larrout  70860 non-null  object
 14  surf     70860 non-null  int64 
 15  infra    70860 non-null  int64 
 16  situ     70860 non-null  int64 
 17  vma      70860 non-null  int64 
dtypes: int64(11), object(7)
memory usage: 9.7+ MB


In [10]:
df_lieux.describe()

Num_Acc          catr            v1          circ          vosp  \
count  7.086000e+04  70860.000000  70860.000000  70860.000000  70860.000000   
mean   2.023000e+11      3.397403     -0.228860      1.712673      0.183164   
std    1.581802e+04      1.113553      0.431536      0.942601      0.744139   
min    2.023000e+11      1.000000     -1.000000     -1.000000     -1.000000   
25%    2.023000e+11      3.000000      0.000000      1.000000      0.000000   
50%    2.023000e+11      3.000000      0.000000      2.000000      0.000000   
75%    2.023000e+11      4.000000      0.000000      2.000000      0.000000   
max    2.023001e+11      9.000000      3.000000      4.000000      3.000000   

               prof         plan          surf         infra          situ  \
count  70860.000000  70860.00000  70860.000000  70860.000000  70860.000000   
mean       1.206336      1.26733      1.254036      0.811177      1.611727   
std        0.534770      0.65518      0.756575      2.162585      1.627796   
min       -1.000000     -1.00000     -1.000000     -1.000000     -1.000000   
25%        1.000000      1.00000      1.000000      0.000000      1.000000   
50%        1.000000      1.00000      1.000000      0.000000      1.000000   
75%        1.000000      1.00000      1.000000      0.000000      1.000000   
max        4.000000      4.00000      9.000000      9.000000      8.000000   

                vma  
count  70860.000000  
mean      53.306393  
std       26.001154  
min       -1.000000  
25%       30.000000  
50%       50.000000  
75%       70.000000  
max      130.000000

In [11]:
df_lieux.isnull().sum()

Num_Acc        0
catr           0
voie       12747
v1             0
v2         64976
circ           0
nbv            0
vosp           0
prof           0
pr             0
pr1            0
plan           0
lartpc     70829
larrout        0
surf           0
infra          0
situ           0
vma            0
dtype: int64

Jeu de données pour les lieux de 70860  lignes pour 18 colonnes.


### Suppression des colonnes inutiles


In [14]:
df_lieux.drop(['voie', 'v1', 'v2', 'pr', 'pr1', 'lartpc'], axis=1, inplace=True)


In [15]:
df_lieux.head()

Num_Acc  catr  circ nbv  vosp  prof  plan larrout  surf  infra  situ  \
0  202300000001     4     1   2     0     1     1      -1     2      0     1   
1  202300000001     4     1   1     0     1     1      -1     2      0     1   
2  202300000002     3     2   3     2     1     1      -1     2      0     1   
3  202300000003     3     2   4     0     1     1      -1     2      5     1   
4  202300000003     3     2   4     0     1     1      -1     2      5     1   

   vma  
0   30  
1   30  
2   50  
3   50  
4   50

### Observation des variables

In [17]:
# Catégorie de routes
df_lieux ['catr'].value_counts(normalize = True)

catr
4    0.456732
3    0.377251
1    0.073130
2    0.056661
7    0.026136
6    0.005828
9    0.003260
5    0.001002
Name: proportion, dtype: float64

In [18]:
# Création de classes
df_lieux['catr'] = df_lieux['catr'].replace(6, 9)
df_lieux['catr'] = df_lieux['catr'].replace(5, 9)

In [19]:
# Catégorie de routes
df_lieux['catr'].value_counts(normalize = True)

catr
4    0.456732
3    0.377251
1    0.073130
2    0.056661
7    0.026136
9    0.010090
Name: proportion, dtype: float64

In [20]:
# Régime de circulation
df_lieux['circ'].value_counts(normalize = True)

circ
 2    0.613647
 1    0.198095
 3    0.111459
-1    0.070858
 4    0.005941
Name: proportion, dtype: float64

In [21]:
# Suppression des -1
df_lieux = df_lieux[df_lieux['circ']!= -1]

In [22]:
# Nombre totale de voie 
df_lieux['nbv'].value_counts(normalize = True)

nbv
2               0.615972
1               0.116496
4               0.088792
3               0.071629
 -1             0.046052
0               0.025046
6               0.017740
5               0.010450
8               0.004314
7               0.001413
#VALEURMULTI    0.000729
10              0.000699
9               0.000334
12              0.000243
11              0.000091
Name: proportion, dtype: float64

In [23]:
# Suppression de la colonne car manque d'infos métiers 
df_lieux.drop('nbv', axis = 1, inplace = True)

C:\Users\peggy\AppData\Local\Temp\ipykernel_25552\3925287845.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_lieux.drop('nbv', axis = 1, inplace = True)


In [24]:
# Existence d'une voie réservée
df_lieux['vosp'].value_counts(normalize = True)

vosp
 0    0.811783
 1    0.057003
-1    0.054800
 2    0.039217
 3    0.037197
Name: proportion, dtype: float64

In [25]:
# Suppression de la colonne car non pertinent
df_lieux.drop('vosp', axis = 1, inplace = True)

C:\Users\peggy\AppData\Local\Temp\ipykernel_25552\3459585061.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_lieux.drop('vosp', axis = 1, inplace = True)


In [26]:
# Rectiligne/courbe
df_lieux['plan'].value_counts(normalize = True)

plan
 1    0.822992
 2    0.086301
 3    0.078601
 4    0.011103
-1    0.001002
Name: proportion, dtype: float64

In [27]:
# Création de classes
df_lieux['plan'] = df_lieux['plan'].replace([3,4], 2)   # En courbe ou en S

C:\Users\peggy\AppData\Local\Temp\ipykernel_25552\3363435687.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_lieux['plan'] = df_lieux['plan'].replace([3,4], 2)   # En courbe ou en S


In [28]:
# Suppression des -1
df_lieux = df_lieux[df_lieux['plan']!= -1]

In [29]:
# Rectiligne/courbe
df_lieux['plan'].value_counts(normalize = True)

plan
1    0.823818
2    0.176182
Name: proportion, dtype: float64

In [30]:
# Largeur chaussée
df_lieux['larrout'].unique()

array([' -1', '12', '3', '14', '5,5', '5', '7', '6', '10', '4', '3,5',
       '10,5', '13,5', '9', '17,5', '17', '0', '21', '6,5', '7,6', '5,7',
       '8', '8,5', '8,56', '4,5', '16', '7,5', '13', '8,1', '4,6', '20,5',
       '3,7', '15', '6,2', '7,72', '5,4', '11', '7,2', '11,5', '9,3',
       '10,75', '7,3', '14,2', '3,8', '6,6', '11,7', '6,32', '6,23',
       '5,8', '6,3', '5,2', '10,6', '6,7', '4,7', '8,2', '5,1', '5,6',
       '6,25', '3,2', '7,29', '6,1', '4,4', '9,5', '7,15', '7,25', '3,25',
       '15,5', '3,6', '1', '8,85', '2,7', '4,3', '7,1', '6,4', '11,1',
       '4,8', '5,3', '7,8', '7,7', '4,75', '5,95', '5,75', '4,2', '9,35',
       '7,11', '18', '3,65', '2,5', '20', '5,44', '7,4', '6,8', '7,65',
       '19', '5,9', '5,13', '2', '7,12', '11,12', '5,35', '3,4', '25',
       '47', '12,7', '2,1', '6,65', '2,85'], dtype=object)

In [31]:
# Suppression de la colonne 'larrout'
df_lieux.drop(['larrout'], axis = 1, inplace = True)

In [32]:
# Etat de la surface 
df_lieux['surf'].value_counts(normalize = True)

surf
 1    0.789883
 2    0.199003
 9    0.003740
 7    0.001824
 3    0.001627
 8    0.001216
 5    0.000973
-1    0.000806
 6    0.000623
 4    0.000304
Name: proportion, dtype: float64

In [33]:
# Création de classes
df_lieux['surf'] = df_lieux['surf'].replace([3,4,6], 2) # Mouillée/flaques/inondées/boue
df_lieux['surf'] = df_lieux['surf'].replace(7, 5)    # Verglas/neige
df_lieux['surf'] = df_lieux['surf'].replace([-1,8], 9)   # Corps gras => autre

In [34]:
# Infrastructure
df_lieux['infra'].value_counts(normalize = True)

infra
 0    0.836149
 5    0.069147
 9    0.029024
 2    0.015447
-1    0.011190
 3    0.010217
 6    0.009077
 1    0.008757
 8    0.007359
 4    0.002980
 7    0.000654
Name: proportion, dtype: float64

In [35]:
# Suppression de la colonne car non pertinent
df_lieux.drop('infra', axis = 1, inplace = True)

In [36]:
# Situation de l'accident
df_lieux['situ'].value_counts(normalize = True)

situ
 1    0.840588
 3    0.061241
 8    0.034133
 5    0.027124
 4    0.020145
 6    0.011327
 2    0.004668
-1    0.000775
Name: proportion, dtype: float64

In [37]:
# Suppression de la colonne car non pertinent
df_lieux.drop('situ', axis = 1, inplace = True)

In [38]:
# Vitesse max autorisée
df_lieux['vma'].value_counts(normalize = True)

vma
 50     0.457437
 30     0.179420
 80     0.123425
 90     0.067961
 70     0.061135
-1      0.060481
 110    0.027914
 130    0.015538
 20     0.002509
 60     0.001809
 40     0.000958
 25     0.000897
 45     0.000167
 100    0.000152
 10     0.000076
 15     0.000030
 75     0.000030
 35     0.000030
 31     0.000015
 65     0.000015
Name: proportion, dtype: float64

In [39]:
# Création de classes
df_lieux['vma'] = df_lieux['vma'].replace([30,20,40,25,45,10,15,35,31,23], 50)
df_lieux['vma'] = df_lieux['vma'].replace([60,65,70,75], 80)
df_lieux['vma'] = df_lieux['vma'].replace(100, 110)

In [40]:
# Suppression des -1
df_lieux = df_lieux[df_lieux['vma']!= -1]

In [41]:
df_lieux.info()

<class 'pandas.core.frame.DataFrame'>
Index: 61795 entries, 0 to 70859
Data columns (total 7 columns):
 #   Column   Non-Null Count  Dtype
---  ------   --------------  -----
 0   Num_Acc  61795 non-null  int64
 1   catr     61795 non-null  int64
 2   circ     61795 non-null  int64
 3   prof     61795 non-null  int64
 4   plan     61795 non-null  int64
 5   surf     61795 non-null  int64
 6   vma      61795 non-null  int64
dtypes: int64(7)
memory usage: 3.8 MB


In [42]:
df_lieux.shape

(61795, 7)

Suppression de 5142 lignes et 10 colonnes

## Exportation du dataframe nettoyé

In [45]:
df_lieux.to_csv('lieux_clean.csv', index=False)